sakura_core/Makefile

```
@@ -1,26 +1,54 @@
 # Makefile for MinGW32/MinGW-W64
 
+# Relative path of "sakura_core" directory from the current directory.
+# Useful when doing out-of-source build.  E.g.
+#   $ cd sakura_core
+#   $ mkdir _build
+#   $ cd _build
+#   $ mingw32-make -f ../Makefile SRCDIR=..
+# Normally, this is derived from the path of Makefile.
+SRCDIR = $(patsubst %/,%,$(dir $(word 1,$(MAKEFILE_LIST))))
```

こう宣言した方が分かりやすくなる気がしました。
```suggestion
SRCDIR = SRCDIR := $(patsubst %/,%,$(subst \,/,$(dir $(firstword $(MAKEFILE_LIST)))))
```
スタック変数 MAKEFILE_LIST の先頭（＝処理中のMakefileパス）を取り出してディレクトリを取得、
取得したディレクトリパスに含まれる '\\' を '/' に置換、
`patsubst` で末尾の `/` を削ったものを `SRCDIR` とする。（ `:=` は強制代入。）

いちおう、mingw64シェルでも実行できました。

---

`:=` を使うかどうかは関係なさそうです。コマンドラインから SRCDIR を指定したらそちらが優先されます。

ま、もともとは `$(MAKEFILE_LIST)` からソースコードディレクトリの位置を取得するという方法が分からなかったので、仕方なく `SRCDIR` をコマンドラインから指定するようにしていただけなので、今となっては `SRCDIR` は基本的にはユーザーには指定させないようにした方がいいのかもしれません。